docs/api_proposal.md

```diff
@@ -0,0 +1,54 @@
+# Taxonomy API proposal
+
+DINA taxonomy API methods. Expanded from [Taxonomy API discussion](https://docs.google.com/spreadsheets/d/1uaJh1qt6mvY0ZCEB6uBKdnUnyGdpQnZcnFqTMHTS5pA/edit#gid=0) google doc.
+
+One goal is to harmonize with existing taxonomy APIs, which benefits downstream users. Makes it more likely that our services can be incorporated into integration projects such as [taxize](https://cran.r-project.org/web/packages/taxize/index.html) R package. 
+
+See [DINA glossary](https://github.com/DINA-Web/dina-use-cases/blob/master/glossary.md) for terms. 
+
+## Assumptions
+
+* there will be some canonical root URL, format to be determined
+* taxonomic names are not guaranteed to be unique, therefore cannot safely perform CRUD options based on name strings without disambiguation 
+* the API will be versioned, and older versions may be deprecated in the future
+
+## Methods
+
+* /taxon : return info about a taxon
+* /search : return potentially matching taxa
+* /mrca : return the most recent common ancestor of two or more taxa
+* /about : return information about the taxonomy
+* /rank : return the rank of a taxon
+* /subtree : return the subtree descended from a taxon
+
+## Questions for discussion
+
+* how much taxonomy editing functionality will we provide through the API?
+** delete a taxon (re-assigning children to parent?)
+** move a taxon (and children)
+** merge taxa
+** rename taxon
+** split taxon
+
+* limits / pagination
```

Comment:
Pagination is covered in jsonapi: http://jsonapi.org/format/#fetching-pagination
and we have been more explicit in https://github.com/DINA-Web/guidelines/blob/master/DINA-Web-API-Guidelines.md
`Pagination MUST be implemented using the offset-based strategy, using the URI paramaters page[offset] and page[limit]`